core/src/main/java/ch/cyberduck/core/ProtocolFactory.java

```diff
@@ -247,6 +274,38 @@ public Protocol forName(final List<Protocol> enabled, final String identifier, f
         return match;
     }
 
+    /**
+     * @param identifier Serialized protocol reference or scheme
+     * @return Matching protocol or default if no match
+     */
+    public Protocol forNameOrDefault(final String identifier) {
+        return forNameOrDefault(identifier, null);
+    }
+
+    /**
+     * @param identifier Serialized protocol reference or scheme
+     * @param provider   Custom inherited protocol definition
+     * @return Matching protocol or default if no match
+     */
+    public Protocol forNameOrDefault(final String identifier, final String provider) {
+        return forNameOrDefault(this.find(), identifier, provider);
+    }
+
+    /**
+     * @param enabled    List of protocols
+     * @param identifier Serialized protocol reference or scheme
+     * @param provider   Custom inherited protocol definition
+     * @return Matching protocol or default if no match
+     */
+    public Protocol forNameOrDefault(final List<Protocol> enabled, final String identifier, final String provider) {
+        final String defaultIdentifier = preferences.getDefault("connection.protocol.default");
+        final Protocol match = forName(enabled, identifier, provider);
+        if(null != match || Objects.equals(defaultIdentifier, identifier)) {
+            return match;
+        }
+        return forName(enabled, identifier, null);
```

Comment:
```suggestion
        return forName(enabled, defaultIdentifier, null);
```

Reply:
Wir müssen noch Testabdeckung dafür haben.

Reply:
Done, forNameOrDefault is 100% covered.